src/game.cpp

```diff
@@ -12592,6 +12592,10 @@ void game::plswim(int x, int y)
         add_msg(_("The water puts out the flames!"));
         u.remove_effect("onfire");
     }
+    if (u.has_effect("onfire")) {
```

Comment:
> I based it off of the 'onfire' effect

So I see (-:
My miraculous glass ball tells me you want to write "glowing" here.

> I marked this WIP because I can't seem to get the glow to actually work.

`player::active_light` seems to be a good point to add the check for the effect on player/NPCs.